WordPress/MediaEditor/MediaEditor.swift

```diff
@@ -53,17 +88,65 @@ public class MediaEditor: UINavigationController {
         present(capability: capabilityEntity, with: image)
     }
 
+    private func cancel() {
+        asyncImage?.cancel()
+        dismiss(animated: true)
+    }
+
     private func present(capability capabilityEntity: MediaEditorCapability.Type, with image: UIImage) {
+        prepareTransition()
+
         let capability = capabilityEntity.init(
             image,
             onFinishEditing: { image, actions in
                 self.onFinishEditing?(image, actions)
-            }, onCancel: {
-                self.dismiss(animated: true)
-            }
+            },
+            onCancel: cancel
         )
         capability.apply(styles: styles)
         currentCapability = capability
+
         pushViewController(capability.viewController, animated: false)
     }
+
+    private func prepareTransition() {
+        let transition: CATransition = CATransition()
+        transition.duration = 0.3
```

Comment:
Magic number: you could use something like `WPAnimationDurationDefault`, or have a custom constant for this class.

Reply:
Fixed in f1cdf6c20b116b69009cb2e69bdde0fc295bead1

Reply:
Marking as resolved.